.github/workflows/test-cygwin.yml

```
@@ -0,0 +1,53 @@
+name: Test Cygwin
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Let's not limit to `main`, and also add a button to trigger builds via `workflow_dispatch`

```suggestion
on: [push, pull_request, workflow_dispatch]
```

---

Done